voting-contract/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_sdk::collections::Map;
+use near_sdk::{env, near_bindgen, AccountId, Balance, BlockHeight, EpochHeight};
+use serde::Serialize;
+use std::collections::{BTreeMap, HashMap};
+
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+const MIN_HEIGHT_HORIZON: u64 = 432000;
+const MAX_HEIGHT_HORIZON: u64 = MIN_HEIGHT_HORIZON * 100;
+
+type ProposalId = u64;
+
+#[derive(BorshDeserialize, BorshSerialize, Eq, PartialEq, Debug, Serialize)]
+pub struct Proposal {
+    /// Proposed height to reset the network.
+    proposed_height: BlockHeight,
```

Comment:
I thought it's a generic contract, and not specific to network resets.

Reply:
So I am slightly confused about that. Do we want to write a general contract that can be used for multiple things? Then we need to encode the actual object that is proposed in some way.

Reply:
Good point. We may want to include binary `metadata: Vec<u8>` field to every proposal, which will keep the required information in a serialized form.

Reply:
@evgenykuzyakov so you want to have one contract that simultaneous hold different types of votes. That seems a bit strange to me and I would like to have one contract per voting, i.e, there will be one contract deployed for network reset height, another one for unlocking.